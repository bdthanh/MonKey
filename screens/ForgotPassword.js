import React, {useState, useEffect} from 'react'
import MainContainer from '../components/Containers/Main';
import KeyboardAvoidingContainer from '../components/Containers/KeyboardAvoiding';
import { Text, View, ActivityIndicator, Alert } from 'react-native';
import styles from '../components/styles';
import TextInputWithIcon from '../components/Containers/TextInputWithIcon';
import LogInButton from '../components/Containers/LogInButton';
import PressableText from '../components/Containers/PressableText';
import { autoNav, handleLogIn, sendResetPasswordEmail } from '../api/authentication';
import SignUp from './Signup';

const ForgotPassword = ({navigation}) => {
  useEffect(() => autoNav(navigation), []);

  const [email, setEmail] = useState('');

  return (
    <MainContainer>
      <KeyboardAvoidingContainer>
        <View style={{alignContent: 'center', paddingBottom: 10}}>
          <Text style={styles.boldBlueText}>Reset your password</Text>
          {/* <Text style={styles.italicText}>Hello, welcome back to your account!</Text> */}
        </View>
          <TextInputWithIcon
            label='Enter your email address'
            icon='mail'
            placeholder='yourmail@gmail.com'
            keyboardType='email-address'
            value={email}
            autoCapitalize = 'none'
            onChangeText={text => setEmail(text)}
          />
          <LogInButton onPress={() => sendResetPasswordEmail(email, navigation)}>Send Password Reset Email</LogInButton>
      </KeyboardAvoidingContainer>
    </MainContainer>
  )
}

export default ForgotPassword;