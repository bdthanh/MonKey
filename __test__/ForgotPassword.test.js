import React from 'react';
import ForgotPassword from '../screens/ForgotPassword';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer
    .create(<ForgotPassword/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});