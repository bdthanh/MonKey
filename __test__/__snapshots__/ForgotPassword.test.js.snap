// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFAEB",
      "flex": 1,
      "padding": 25,
      "paddingTop": 50,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "flex": 1,
        },
        Object {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <RCTScrollView
      showsVerticalScrollIndicator={false}
    >
      <View>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <View
            style={
              Object {
                "alignContent": "center",
                "paddingBottom": 20,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#35baf6",
                  "fontSize": 42,
                  "fontWeight": "bold",
                  "marginBottom": 10,
                }
              }
            >
              Reset your password
            </Text>
          </View>
          <View>
            <View
              style={
                Object {
                  "borderRightColor": "#35baf6",
                  "borderRightWidth": 2,
                  "left": 13,
                  "paddingRight": 8,
                  "position": "absolute",
                  "top": 40,
                  "zIndex": 1,
                }
              }
            >
              <Text />
            </View>
            <Text
              style={
                Object {
                  "color": "#35baf6",
                  "fontSize": 18,
                  "fontWeight": "400",
                }
              }
            >
              Enter your email address
            </Text>
            <TextInput
              autoCapitalize="none"
              keyboardType="email-address"
              onChangeText={[Function]}
              placeholder="yourmail@gmail.com"
              placeholderTextColor="#ADD8E6"
              style={
                Object {
                  "backgroundColor": "#FFFAEB",
                  "borderColor": "#35baf6",
                  "borderRadius": 8,
                  "borderWidth": 2,
                  "color": "#35baf6",
                  "fontSize": 18,
                  "height": 60,
                  "marginBottom": 25,
                  "marginTop": 3,
                  "padding": 15,
                  "paddingLeft": 60,
                  "paddingRight": 55,
                }
              }
              value=""
            />
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#35baf6",
                "borderRadius": 10,
                "flex": 1,
                "flexDirection": "row",
                "height": 55,
                "justifyContent": "center",
                "opacity": 1,
                "padding": 15,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#FFFAEB",
                  "fontSize": 21,
                  "fontWeight": "500",
                }
              }
            >
              Send Password Reset Email
            </Text>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignSelf": "center",
                "paddingTop": 10,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#35baf6",
                  "fontSize": 18,
                  "fontWeight": "400",
                }
              }
            >
              Back to login page
            </Text>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
